app/models/task.py

```diff
@@ -1,5 +1,36 @@
-from sqlalchemy.orm import Mapped, mapped_column
+from sqlalchemy.orm import Mapped, mapped_column, relationship
+from datetime import datetime
 from ..db import db
+from typing import Optional
 
-class Task(db.Model):
+class Task(db.Model): # Declares model Task, tells SQLAlchemy to map Task to db table
     id: Mapped[int] = mapped_column(primary_key=True, autoincrement=True)
+    title: Mapped[str]
+    description: Mapped[str]
+    completed_at: Mapped[datetime] = mapped_column(nullable=True)
+
+    goal_id: Mapped[Optional[int]] = mapped_column(db.ForeignKey("goal.id"), nullable=True) # Adds foreign key reference to goal.id, each task can be linked to one goal
+    goal: Mapped[Optional["Goal"]] = relationship(back_populates="tasks") # Sets up many-to-one-relat*p from task to Goal
```

Comment:
Nice job using `Optional[]` to indicate that these values are nullable. When we use `Optional[]`, we don't need `nullable=True` as an argument for `mapped_column`. Review SQLAlchemy docs linked in my comment above to read more about how the library works if this concept still feels a little fuzzy. 

```suggestion
    goal_id: Mapped[Optional[int]] = mapped_column(db.ForeignKey("goal.id")) # Adds foreign key reference to goal.id, each task can be linked to one goal
```

Reply:
Fixed as requested.